files/image_config/hostcfgd/hostcfgd

```diff
@@ -367,14 +374,19 @@ class HostConfigDaemon:
             self.update_feature_state(feature_name, state, feature_table)
 
     def start(self):
-        # Update all feature states once upon starting
-        self.update_all_feature_states()
 
         self.config_db.subscribe('AAA', lambda table, key, data: self.aaa_handler(key, data))
         self.config_db.subscribe('TACPLUS_SERVER', lambda table, key, data: self.tacacs_server_handler(key, data))
         self.config_db.subscribe('TACPLUS', lambda table, key, data: self.tacacs_global_handler(key, data))
         self.config_db.subscribe('LOOPBACK_INTERFACE', lambda table, key, data: self.lpbk_handler(key, data))
         self.config_db.subscribe('FEATURE', lambda table, key, data: self.feature_state_handler(key, data))
+
+        # Update all feature states once upon starting
+        self.update_all_feature_states()
+
+        # Defer load until subscribe
+        self.load()
+
         self.config_db.listen()
```

Comment:
I see the code, find that self.config_db.subscribe() is a misnomer .. it doesn't actually do a pubsub.psubscribe(). It just assigns a handler and comes out. It is this API self.config_db.listen() which actually does pubsub.psubscribe()

So we could still miss the events since self.update_all_feature_states() is before listen call ?  
Just wondering if this fix will take care of all cases still ?

Reply:
@judyjoseph   self.load() should take care  care of that at least for TACACS/AAA
I think one window is config-db write happening in between self.load() and self.config_db.listen(). Might be very corner case.

Reply:
Still this 23 seconds long call -- taken to start all services, even when all are running, is way too expensive. We would need to take a second look at it. Better would be to run it on a different thread using timer.

Reply:
yes !! It may be ok for the user to wait for 23 secs to get login access in case of single ASIC. On multi-asic wait will be longer, as we have more  processes per ASIC and the total wait could be more than 1min.